File: src/main/java/com/jake/kotlinjava/todo/controller/TodoController.java
package com.jake.kotlinjava.todo.controller;

import com.jake.kotlinjava.todo.controller.model.TodoListResponse;
import com.jake.kotlinjava.todo.controller.model.TodoRequest;
import com.jake.kotlinjava.todo.controller.model.TodoResponse;
import com.jake.kotlinjava.todo.domain.Todo;
import com.jake.kotlinjava.todo.service.TodoService;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping(value = "/api/todos", produces = MediaType.APPLICATION_JSON_VALUE)
public class TodoController {

	final TodoService todoService;

	public TodoController(TodoService todoService) {
		this.todoService = todoService;
	}

	@GetMapping
	public ResponseEntity<TodoListResponse> getAll() {
		List<Todo> todos = todoService.findAll();
		return ResponseEntity.ok(TodoListResponse.of(todos));
	}

	@GetMapping("/{id}")
	public ResponseEntity<TodoResponse> get(@PathVariable Long id) {
		Todo todo = todoService.findById(id);
		return ResponseEntity.ok(TodoResponse.of(todo));
	}

	@PostMapping
	public ResponseEntity<TodoResponse> create(@RequestBody TodoRequest request) {
		Todo todo = todoService.create(request);
		return ResponseEntity.ok(TodoResponse.of(todo));
	}

	@PutMapping("/{id}")
	public ResponseEntity<TodoResponse> update(@PathVariable Long id, @RequestBody TodoRequest request) {
		Todo todo = todoService.update(id, request);
		return ResponseEntity.ok(TodoResponse.of(todo));
	}

	@DeleteMapping("/{id}")
	public ResponseEntity<Void> delete(@PathVariable Long id) {
		todoService.delete(id);
		return ResponseEntity.noContent().build();
	}
}

File: src/main/java/com/jake/kotlinjava/todo/controller/TodoController.kt
package com.jake.kotlinjava.todo.controller

import com.jake.kotlinjava.todo.controller.model.TodoListResponse
import com.jake.kotlinjava.todo.controller.model.TodoRequest
import com.jake.kotlinjava.todo.controller.model.TodoResponse
import com.jake.kotlinjava.todo.service.TodoService
import org.springframework.http.MediaType
import org.springframework.http.ResponseEntity
import org.springframework.http.ResponseEntity.*
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(value = ["/api/todos"], produces = [MediaType.APPLICATION_JSON_VALUE])
class TodoController(
    private val todoService: TodoService
) {

    @GetMapping
    fun getAll() =
        ok(TodoListResponse.of(todoService.findAll()))

    @GetMapping("/{id}")
    fun get(@PathVariable id: Long) =
        ok(TodoResponse.of(todoService.findById(id)))

    @PostMapping
    fun create(@RequestBody request: TodoRequest) =
        ok(TodoResponse.of(todoService.create(request)))

    @PutMapping("/{id}")
    fun update(@PathVariable id: Long, @RequestBody request: TodoRequest) =
        ok(TodoResponse.of(todoService.update(id, request)))

    @DeleteMapping("/{id}")
    fun delete(@PathVariable id: Long): ResponseEntity<Unit> {
        todoService.delete(id)
        return noContent().build()
    }

}